### PR TITLE
Don't retry network errors for non-idempotent requests

### PR DESCRIPTION
**Pull request by robfletcher**
This fixes an issue where we'll retry things like create server group on CloudDriver if we get a network error. We should only ever retry things that are known to be idempotent.

### CODE REVIEW

**Comment by robfletcher** [comment]
PTAL @spinnaker/netflix-reviewers

**Comment by robfletcher** [comment]
SPIN-1623

**Comment by ajordens** [comment]
I dig it, LGTM

File: orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy [hunk 1]
```diff
@@ -56,8 +60,31 @@ class RetrofitExceptionHandler implements ExceptionHandler<RetrofitError> {
       response.details.kind = e.kind
       response.details.status = properties.status ?: null
       response.details.url = properties.url ?: null
-      response.shouldRetry = (e.kind == RetrofitError.Kind.NETWORK)
+      response.shouldRetry = (e.kind == NETWORK && findHttpMethodAnnotation(e) in ["GET", "HEAD"])
```
Comment (cfieber) [inline orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy]: I think you could add PUT and DELETE here - they should be idempotent ( I'm sure there's somewhere we are doin it wrong but nothing comes to mind )

Comment (robfletcher) [inline orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy]: Yeah, I was wondering about DELETE specifically. I think the problem with PUT is it gets used wrong as often as it gets used right. I know I mix up PUT and POST all the time.

Comment (robfletcher) [inline orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy]: If it comes to that I guess PATCH could also be considered idempotent.

**Pull request merged**